src/communication_interface.cc

```diff
@@ -412,6 +412,13 @@ void CommunicationInterface::HandlePause(
   bool desired_pause {pause_cmd_msg->data};
   auto source {pause_cmd_msg->source};
 
+  if (source == "cancel_plan") {
+    dexai::log()->error(
+        "CommunicationInterface::HandlePause: Received cancel plan request!");
+    cancel_plan_requested_ = true;
+    return;
+  }
+
```

Comment:
use the value of the int `data` for checking cancellation, not the `source` which should describe the source